src/dos/dos_mscdex.cpp

```diff
@@ -285,7 +285,8 @@ int CMscdex::AddDrive(Bit16u _drive, char* physicalPath, Bit8u& subUnit)
 		Bit16u driverSize = sizeof(DOS_DeviceHeader::sDeviceHeader) + 10; // 10 = Bytes for 3 callbacks
 		
 		// Create Device Header
-		Bit16u seg = DOS_GetMemory(driverSize/16+((driverSize%16)>0));
+		assert((driverSize % 16) == 0); // should always be zero
```

Comment:
Nice find :) We can make it a compile-time check this way:

```
constexpr uint16_t driverSize = sizeof(DOS_DeviceHeader::sDeviceHeader) + 10; // 10 = Bytes for 3 callbacks
// Create Device Header
static_assert((driverSize % 16) == 0); // should always be zero
```

Reply:
Ah, sorry, C++11 requires a message in static assert:

    static_assert((driverSize % 16) == 0, "should always be zero");

Reply:
Darn; I can't replace either `assert` with `static_assert` because neither can be evaluated at compile-time. I suppose the PVS analyzer can dig a bit deeper than the compiler in figuring these out during static analysis. 

```
dos_mscdex.cpp: In member function ‘int CMscdex::AddDrive(Bit16u, char*, Bit8u&)’:
dos_mscdex.cpp:288:35: error: non-constant condition for static assertion
  288 |   static_assert((driverSize % 16) == 0, "should always be zero");
      |                 ~~~~~~~~~~~~~~~~~~^~~~
dos_mscdex.cpp:288:35: error: the value of ‘driverSize’ is not usable in a constant expression
dos_mscdex.cpp:285:10: note: ‘Bit16u driverSize’ is not const
```


```
dos_mscdex.cpp: In member function ‘bool CMscdex::GetDirectoryEntry(Bit16u, bool, PhysPt, PhysPt, Bit16u&)’:
dos_mscdex.cpp:714:32: error: non-constant condition for static assertion
  714 |      static_assert(entryLength <= 256, "entryLength should never exceed 256");
      |                    ~~~~~~~~~~~~^~~~~~
dos_mscdex.cpp:714:32: error: the value of ‘entryLength’ is not usable in a constant expression
```

Reply:
Look at my first comment: `constexpr uint16_t driverSize = …` will solve that.

Reply:
Added `assertm` as a  #define to `support.h` to make it easier to include a message, per the example here. https://en.cppreference.com/w/cpp/error/assert

Works as expected - 

```
d: dos_mscdex.cpp:289: int CMscdex::AddDrive(Bit16u, char*, Bit8u&): Assertion `((void)"should always be zero", (driverSize % 16) == 0)' failed.
```

Reply:
The new asserts now use `assertm` (force pushed and workflow underway)

Reply:
Sorry @dreamer ; `constexpr uint16_t` did the trick, and is indeed static.
Pushing again! Thank you for the keen eye and reminder :-)